internal/oci/runtime_oci.go

```diff
@@ -228,14 +228,17 @@ func (r *runtimeOCI) CreateContainer(c *Container, cgroupParent string) (err err
 // StartContainer starts a container.
 func (r *runtimeOCI) StartContainer(c *Container) error {
 	c.opLock.Lock()
-	defer c.opLock.Unlock()
 
 	if _, err := utils.ExecCmd(
 		r.path, rootFlag, r.root, "start", c.id,
 	); err != nil {
+		c.opLock.Unlock()
 		return err
 	}
+	c.opLock.Unlock()
```

Comment:
This should still remain locked for the duration of the function.

Reply:
Addressed